### PR TITLE
Fix #1745 search box breaking to multiple lines 

### LINKED ISSUE #1745: Mobile search box breaks with international languages
When the search category label is so long on mobile, the search box breaks to multiple lines, this is the case for some languages.
image
and other problem with translation not updating
ezgif com-gif-maker

### PR DESCRIPTION
**Pull request by Zaydme**
### What does this pull request aim to achieve? 💯
When the search category label is so long on mobile, the search box breaks to multiple lines, this is the case for some languages.
image

and other problem with translation not updating
ezgif com-gif-maker

_Please be concise and link any related issue(s):_
#1745

- [x] I have checked that no one else is working on similar changes.
- [x] I have read the latest [**README**](README.md) and understand the development workflow.
- [x] The name of this branch is: **`feature/<branch_name>`**, or **`hotfix/<branch_name>`** if this is a hotfix, where `<branch_name>` briefly describes the issue(s) resolved and is prefixed with the issue number(s) (e.g. `1127-1130-update-git-branching-model`). The branch is created in a cloned version of this repo, **not a forked repo**.
- [x] This branch is rebased/merged with the **latest staging** (or the **latest production** if it is a hotfix).
- [x] There are no merge conflicts.
- [x] There are no console warnings and errors.
- [x] On the right hand side of this PR, I have linked the appropriate issue(s) under "Issues" and added a project progress under "Projects".
- [x] The title describes the issue(s) being addressed using format: **`<issue_numbers> - <brief_issue_description>`** (e.g. `#1127/#1130 - Update Git Branching Model`)

### CODE REVIEW

**Review by rabbi08 — CHANGES_REQUESTED** [review]
@Zaydme 
Tested on iPhone 11 Pro Max, though UI issue not observed but new issue have been observed.

The searched keyword is being replaced by the first TAG, this issue do exist in STG. please below screenshot.

image

**Comment by Zaydme** [comment]
@rabbi08 do you mean when you start typing? 
For example for English we have "Posts" "Organization" "People".
- if you type: "Po" it will automatically select "Posts" and empty your text box.
- if you type: "Or" it will automatically select "Organization" and empty your text box.
- if you type: "Pe" it will automatically select "People" and empty your text box.
- or if you type something like "He" it will automatically select "Posts" but it will keep yout text, so it will be "`Posts: `He"

**Comment by rabbi08** [comment]
@Zaydme 
The "post" term is overwritten, please see below GIF

image

**Comment by Zaydme** [comment]
@rabbi08 
do you mean this cut here? which is normal when the text is longer than the text field
image
or was that related to this issue where the search terms weren't getting translated unless you refresh the page?
(it was fixed in the last commit)
ezgif com-gif-maker

**Comment by rabbi08** [comment]
It's not related to translation it is overwritten issue of the Post with the searched item.

**Comment by rabbi08** [comment]
> It's not related to translation it is overwritten issue of the Post with the searched item.

@Zaydme 
sorry for confusion. 
Though you have mention this too but let me rephrase here, the translation is not kicked in when the page is translated after changing the language from the dropdown. The translation is Triger after user navigates from other Tab/page. Let me know if that make sense.

**Comment by rabbi08** [comment]
Please see this GiF as the translation fir the tabs trigger after user navigate to profile page and then comes to Helpboard . The TABs are translated after that

image

**Comment by rabbi08** [comment]
@Zaydme 
To remove previous confusion I tested again and it looks that the Mobile UI issue and the translation issue are working fine, but there is 500 loading error after the search is being performed. Please see below GIF 
image

image

**Comment by Zaydme** [comment]
> Please see this GiF as the translation fir the tabs trigger after user navigate to profile page and then comes to Helpboard . The TABs are translated after that

@rabbi08 
yes this behavior was changed with this PR, translation should update immediately.
just like [this](https://github.com/FightPandemics/FightPandemics/pull/1746#issuecomment-720703653) :+1: 

For the 500 errors, I think there is a review branch that haven't pulled the search code yet, and it's causing this problem, @mannykary did what I said make sense?
but PROD have it's own database so that shouldn't be the case there.

**Comment by rabbi08** [comment]
> > Please see this GiF as the translation fir the tabs trigger after user navigate to profile page and then comes to Helpboard . The TABs are translated after that
> 
> @rabbi08
> yes this behavior was changed with this PR, translation should update immediately.
> just like [this](https://github.com/FightPandemics/FightPandemics/pull/1746#issuecomment-720703653) 👍
> 
> For the 500 errors, I think there is a review branch that haven't pulled the search code yet, and it's causing this problem, @mannykary did what I said make sense?
> but PROD have it's own database so that shouldn't be the case there.

Agree @Zaydme 
So we can merge the code in STG for final testing there.

**Review by rabbi08 — APPROVED** [review]
As the mentioned issues are fixed in the feature branch and are working fine. We can merge this PR in STG for end-to-end test. Thanks
